src/components/ui/SnapCarousel/index.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { createContext, useContext, useMemo, useRef } from 'react';
+import styled from 'styled-components';
+
+import { noop } from '../../../utils';
+
+const Container = styled.div({
+  // overscrollBehavior: 'none' // it prevents scroll-y in carousel view
+  zIndex: 24,
+  display: 'flex',
+  flexDirection: 'row',
+  scrollSnapType: 'x mandatory',
+  overflowY: 'scroll',
+  gap: 12,
+  scrollPadding: 0,
+  paddingLeft: 0,
+  scrollbarWidth: 'none',
+  userSelect: 'none',
+  '::-webkit-scrollbar': {
+    display: 'none',
+  },
+});
+
+const ItemContainer = styled.div({
+  display: 'flex',
+  flexShrink: 0,
+  scrollSnapAlign: 'start',
+  flexDirection: 'column',
+  borderRadius: 16,
+  overflow: 'hidden',
+  border: '1px solid var(--sendbird-light-onlight-04)',
+  backgroundColor: '#fff',
+  boxSizing: 'border-box',
+  cursor: 'pointer',
+});
+
+type SnapCarouselProps = {
+  width?: number | string;
+  height?: number | string;
+  children: React.ReactNode;
+  gap?: number;
+  startPadding?: number;
+  style?: React.CSSProperties;
+};
+
+const Context = createContext<{
+  activeIndex: React.MutableRefObject<number>;
+  scrollTo: (index: number) => void;
+}>({ activeIndex: { current: 0 }, scrollTo: noop });
+
+export const SnapCarousel = ({
+  gap = 0,
+  startPadding = 0,
+  style,
+  children,
+}: SnapCarouselProps) => {
+  const ref = useRef<HTMLDivElement>(null);
+  const activeIndex = useRef(0);
+  const itemLength = React.Children.toArray(children).length;
+  const cursorSize = useMemo(() => {
+    const containerWidth = (ref.current?.scrollWidth ?? 0) - startPadding;
+    return containerWidth / itemLength + gap;
+  }, [ref.current?.scrollWidth, itemLength, gap, startPadding]);
+
+  const onScroll = (e: React.UIEvent<HTMLDivElement>) => {
+    const idx = Math.round(e.currentTarget.scrollLeft / cursorSize);
+    if (idx !== activeIndex.current) activeIndex.current = idx;
+  };
+
+  const scrollTo = (index: number) => {
+    if (ref.current) {
+      activeIndex.current = Math.min(Math.max(0, index), itemLength - 1);
+      ref.current.scroll({
+        left: activeIndex.current * cursorSize,
```

Comment:
when index is the last item, snapping position seems little off (padding not considered). Is this intended?
![thisone](https://github.com/sendbird/chat-ai-widget/assets/16806397/92398ae0-a12e-4c9c-83ae-e8c9df72a74b)

Reply:
It's still under development.
You can check out the latest UI in this video.

https://github.com/sendbird/chat-ai-widget/assets/26326015/a592f4d3-813c-4428-9e33-1e66d628f716